### DISCUSSION TITLE
The internal Heading ID refernece links won't link to the location in PDF but to the Website link.

### DISCUSSION THREAD

**Discussion by GretaSuper** [facebook/docusaurus#9615] [Q&A]
My project has used the docusaurus for user-guide and will convert it to PDF in the end. 

The chapter hierachy will look like it:
```
---
id: Chapter5
title: 5 Chapter_Name
---

## 5.3. Level2_section_title {#5.3}
### 5.3.1. Level3_section_title {#5.3.1}
#### 5.3.1.2 Level4_section_title {#5.3.1.2}
```

And I would need to refer to the above heading ID at another chapter.

As I used one of NPM tools called "docs-to-pdf" to convert my document to PDF. (Ref: https://socket.dev/npm/package/docs-to-pdf). 
It seems the reference usage won't really redirect to the designated link in the PDF document but pop up with Internet page. 

I use the following usage and neither of them can solve my problems.
```
1. [Apeednfix H](#AppendixH)
2. <Link to="#AppendixH">Appendix H</Link>
3. <a href="AppendixH"> Appendix H. How to Sign a Driver in Linux:</a>
```
My docs-to-pdf command is as below: 

```
npx docs-to-pdf 
--initialDocURLs="http://localhost:3000/docs/xxxx" 
--contentSelector="article" 
--paginationSelector="a.pagination-nav__link.pagination-nav__link--next" 
--outputPDFFilename="xxxx.pdf" 
--pdfMargin="96,79,96,96"
--coverSub='<html>......</html>' 
--headerTemplate="<div style='...</div>" 
 --footerTemplate="<div style=....</div>"
```

Does any expert have this problem as me before? 

Need help and your support is a lot appreciated.

Thanks!

**Comment by slorber** [comment]
I am not able to understand your problem sorry.
Please assume I know nothing about docs-to-pdf

**Comment by Josh-Cena** [comment]
The problem is that all links we generate are fully resolved. For any anchor link with href `#foo`, we always prepend the pathname so it because `/foo/bar#foo`. I don't think we want to change that, because this is a totally normal thing to do; you would want to ask docs-to-pdf to automatically resolve pathname hrefs to PDF locations.
